product_variant_configurator/models/pricelist.py

```diff
@@ -0,0 +1,200 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Oihane Crucelaegui - AvanzOSC
+# Copyright 2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# Copyright 2016 ACSONE SA/NV
+# Copyright 2017 David Vidal <david.vidal@tecnativa.com>
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3
+
+from itertools import chain
+from odoo import api, fields, models, tools
+from odoo.exceptions import UserError
+
+
+class ProductPricelist(models.Model):
+    _inherit = 'product.pricelist'
+
+    @api.multi
+    def _compute_price_rule(self, products_qty_partner, date=False,
+                            uom_id=False):
+        """ Low-level method - Mono pricelist, multi products
+        Returns: dict{product_id: (price, suitable_rule) for the pricelist}
+
+        If date in context: Date of the pricelist (%Y-%m-%d)
+
+            :param products_qty_partner: list of typles products, quantity,
+            partner
+            :param datetime date: validity date
+            :param ID uom_id: intermediate unit of measure
+        """
+        self.ensure_one()
+        if not date:
+            date = self._context.get('date', fields.Date.today())
+        if not uom_id and self._context.get('uom'):
+            uom_id = self._context['uom']
+        if uom_id:
+            # rebrowse with uom if given
+            # TODO: Avoid rewrite whole method for this sole line:
```

Comment:
@pedrobaeza  Yes, I put that TODO myself as the old methods are deprecated and I had to implement the new ones. I really like your method's idea but it still falls on an infinite recursion in the last return because the first element of the tuple `product_qty_partner` is going to always be a `product.template` object (the first check should call to super)